client/mm/libxc/binance.go

```diff
@@ -1915,7 +1915,34 @@ func binanceMarketToDexMarkets(binanceBaseSymbol, binanceQuoteSymbol string, tok
 	return markets
 }
 
+type BNApiResponse struct {
+	Code int    `json:"code"`
+	Msg  string `json:"msg"`
+}
+
 func requestInto(req *http.Request, thing interface{}) error {
-	// bnc.log.Tracef("Sending request: %+v", req)
-	return dexnet.Do(req, thing, dexnet.WithSizeLimit(1<<24))
+	var sizeLimit int64 = 1 << 24
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("error performing request: %w", err)
+	}
+	defer resp.Body.Close()
+	// https://binance-docs.github.io/apidocs/websocket_api/en/#response-format
+	if resp.StatusCode != http.StatusOK {
+		var apiResp BNApiResponse
+		reader := io.LimitReader(resp.Body, sizeLimit)
+		if err = json.NewDecoder(reader).Decode(&apiResp); err != nil {
+			return fmt.Errorf("error decoding response: %w", err)
+		}
+		return fmt.Errorf("API error %d: %s (%d)", resp.StatusCode, apiResp.Msg, apiResp.Code)
+	}
```

Comment:
Let's just add this functionality to dexnet. https://github.com/decred/dcrdex/compare/50005f4...buck54321:dexnet-errparse